filequeue

swiftgist opened the pull request:
Change parameter, add unit tests

jan--f commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,172 @@
+        touched = fq.touch("blue")
+        files = fq.ls()
+        shutil.rmtree(dirpath)
+        assert files == ['red', 'blue']
```
Getting a test failure here.
```
assert ['blue', 'red'] == ['red', 'blue']
```

So its just an ordering issue. Either sort both lists before the comparison or compare sets.

jschmid1 commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,172 @@
+        fq = filequeue.FileQueue(root_dir=dirpath)
+        queue = os.path.isdir("{}/default".format(dirpath))
+        shutil.rmtree(dirpath)
+        assert queue == True
```
comparison to True should be done with 'is'

jschmid1 commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,172 @@
+        Verify that the default queue exists
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath)
```
fq is assigned but never used

jschmid1 commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,172 @@
+        '''
+        dirpath = tempfile.mkdtemp()
+        fq = filequeue.FileQueue(root_dir=dirpath)
+        touched = fq.touch("red")
```
and no need to assign 'touched'

jschmid1 commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,172 @@
+        fq = filequeue.FileQueue(root_dir=dirpath, queue='lunch')
+        queue = os.path.isdir("{}/lunch".format(dirpath))
+        shutil.rmtree(dirpath)
+        assert queue == True
```
same as: https://github.com/SUSE/DeepSea/pull/137/files#r105359743

jschmid1 commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,172 @@
+        '''
+        Verify that the default queue exists
+        '''
+        dirpath = tempfile.mkdtemp()
```
instead of wirting `dirpath = tempfile.mkdtemp()` in every method, pytest suggests to use _fixtures_

http://doc.pytest.org/en/latest/fixture.html

jschmid1 replied:
same can be done for `filequeue.FileQueue(root_dir=dirpath)`

swiftgist commented:
Fixed the ordering issue and added a fixture.  

It seems like a toss up to me with respect to a fixture in this case.  While it saves a few lines, I am adding a new parameter to each function.  I think the amount of time to figure out either is about the same.  If the fixture has a few steps, I can see the advantage.

If this were more complicated, I could setup a context or wrapper class that would preserve dirpath between the setup and teardown.  It seems though pytest is considering that method to be the "classical" way.  However, I do not think adding any of this will improve readability and would only increase complexity in this instance.

jan--f commented on tests/unit/runners/test_filequeue.py:
```diff
@@ -0,0 +1,166 @@
+        touched = fq.touch("blue")
+        files = fq.ls()
+        shutil.rmtree(dirpath)
+        assert set(files) == set(['red', 'blue'])
```
The ordering makes this test fail still.
